staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -300,21 +300,20 @@ func (s *store) GuaranteedUpdate(
 		}
 
 		ret, ttl, err := s.updateState(origState, tryUpdate)
-		if err != nil {
-			// It's possible we were working with stale data
-			if mustCheckData && apierrors.IsConflict(err) {
-				// Actually fetch
-				origState, err = getCurrentState()
-				if err != nil {
-					return err
-				}
-				mustCheckData = false
-				// Retry
-				continue
-			}
-
+		if err != nil && !mustCheckData {
 			return err
 		}
+		if err != nil && mustCheckData {
```

Comment:
`(err != nil && mustCheckData) || apierrors.IsConflict(err)` to still ensure we retry up to max retries if there is write contention?

Reply:
Discussed offline, the write contention is retried here: https://github.com/kubernetes/kubernetes/blob/71bbabc36b78212d74e77285b5a0b51170a8eded/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L357-L377

The reason that the [original code (https://github.com/kubernetes/kubernetes/blob/71bbabc36b78212d74e77285b5a0b51170a8eded/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L305) checks for apierror.IsConflict(err) is not to retry on write contention, but is to make sure the error emitted by this [line](https://github.com/kubernetes/kubernetes/blob/71bbabc36b78212d74e77285b5a0b51170a8eded/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L523) is retried with fresh data.